logos/logos-agent/app/sessions.py

```
@@ -1493,7 +1505,46 @@ async def _may_try_again(session: dict[str, Any]) -> bool:
+        attempt with the failure in its task.
+        """
+        branch = str(session.get("branch_name") or "")
+        if not branch or not session.get("trigger_ref"):
+            # A session queued by a person is that person's to follow up.
+            return
+        asyncio.create_task(self._take_up_a_red_build(session, branch, pushed_sha))
+
+    async def _take_up_a_red_build(self, session: dict[str, Any], branch: str, pushed_sha: str) -> None:
+        try:
+            status, detail = await github.wait_for_pr_builds(branch, pushed_sha)
+        except Exception as exc:
```

Ruff (0.16.3)

[warning] Do not catch blind exception: `Exception`

(BLE001)